internal/broadcaster/utxo_creator_test.go

```diff
@@ -1,83 +1,110 @@
+			getBalanceFunc: func(ctx context.Context, address string, constantBackoff time.Duration, retries uint64) (int64, int64, error) {
+				return 400, 0, nil
 			},
-			expectedError:      nil, // Start should handle zero outputs gracefully
-			expectedStartCalls: 1,
+			expectedBroadcastCalls: 1,
+			expectedError:          nil,
+			requestedUTXOs:         4,
+			requestedAmountPerUTXO: 100,
 		},
 		{
-			name:              "invalid input - zero satoshis per output",
-			outputs:           5,
-			satoshisPerOutput: 0,
-			startFunc: func(outputs int, satoshisPerOutput uint64) error {
-				return nil // Simulate success to test input handling
+			name:         "Insufficient balance",
+			getUTXOsResp: sdkTx.UTXOs{{TxID: testdata.TX1Hash[:], Vout: 0, LockingScript: ks.Script, Satoshis: 50}},
+			getBalanceFunc: func(ctx context.Context, address string, constantBackoff time.Duration, retries uint64) (int64, int64, error) {
+				return 100, 0, nil
 			},
-			expectedError:      nil, // Start should handle zero satoshis gracefully
-			expectedStartCalls: 1,
+			expectedBroadcastCalls: 0,
+			expectedError:          errors.New("requested total of satoshis exceeds balance\nrequested: 400, balance: 100"),
+			requestedUTXOs:         4,
+			requestedAmountPerUTXO: 100,
 		},
 	}
 
-	for _, tc := range tt {
-		t.Run(tc.name, func(t *testing.T) {
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
 			// Given
-			mockedCreator := &mocks.CreatorMock{
-				StartFunc: tc.startFunc,
+			mockArcClient := &mocks.ArcClientMock{
+				BroadcastTransactionsFunc: func(ctx context.Context, txs sdkTx.Transactions, waitForStatus metamorph_api.Status, callbackURL, callbackToken string, fullStatusUpdates, skipFeeValidation bool) ([]*metamorph_api.TransactionStatus, error) {
+					statuses := make([]*metamorph_api.TransactionStatus, len(txs))
+					for i, tx := range txs {
+						statuses[i] = &metamorph_api.TransactionStatus{Txid: tx.TxID(), Status: metamorph_api.Status_SEEN_ON_NETWORK}
+					}
+					return statuses, nil
+				},
+			}
+
+			mockUtxoClient := &mocks.UtxoClientMock{
+				GetBalanceWithRetriesFunc: tt.getBalanceFunc,
+				GetUTXOsWithRetriesFunc: func(ctx context.Context, lockingScript *script.Script, address string, constantBackoff time.Duration, retries uint64) (sdkTx.UTXOs, error) {
+					return tt.getUTXOsResp, nil
+				},
 			}
+
+			utxoCreator, err := broadcaster.NewUTXOCreator(logger, mockArcClient, ks, mockUtxoClient, false)
+			require.NoError(t, err)
+			defer utxoCreator.Shutdown()
+
 			// When
-			err := mockedCreator.Start(tc.outputs, tc.satoshisPerOutput)
+			actualError := utxoCreator.Start(tt.requestedUTXOs, tt.requestedAmountPerUTXO)
 
 			// Then
-			if tc.expectedError != nil {
-				require.Error(t, err)
-				require.EqualError(t, err, tc.expectedError.Error())
+			if tt.expectedError != nil {
+				require.EqualError(t, actualError, tt.expectedError.Error(), "Error message mismatch")
```

Comment:
Here you can compare the actual errors instead of the error strings
```go
require.ErrorIs(t, actualError, tt.expectedError)
```

Reply:
i have pushed the changes